lib/fs/promises.js

```diff
@@ -147,15 +147,17 @@ async function readFileHandle(filehandle, options) {
 
   const chunks = [];
   const chunkSize = Math.min(size, 16384);
-  const buf = Buffer.alloc(chunkSize);
   let totalRead = 0;
+  let endOfFile;
```

Comment:
Initialize this to `false`.

Reply:
Why though?

Reply:
Better to keep one variable the same type (undefined vs. boolean), when it is used as a boolean anyway.

Reply:
Fair enough